omod/src/main/webapp/resources/scripts/metadata/manageRoles.js

```diff
@@ -122,12 +122,27 @@ angular.module("manageRoles", [ "roleService", "privilegeService", "ngDialog", "
                         $scope.privileges = results;
                         $scope.inheritedPrivilegeFlags = Array($scope.privileges.length);
                         $scope.privilegeFlags = Array($scope.privileges.length);
-                                                
+                        filterPrivileges();                       
                         loadInheritedPrivileges();
 
                     });
                 });
             }
+            //Filters Privileges to required criteria. ie Only "Application privileges" are required.
+            function filterPrivileges(){
+            	var inheritedPrivileges =[];
+            	$scope.privileges.forEach(function(val, idx) {
+            		if($scope.privileges[idx].name.indexOf(":")>-1){
```

Comment:
Don't you only need to check if the privilege has a **Privilege:** prefix?

Reply:
@wluyima  you know am kinda of a newbie, but I thought this checks for presence of the prefix ":"
```
if($scope.privileges[idx].name.indexOf(":")>-1){
```
Or I was required to do it in a cleaner way?

Reply:
Or did you mean checking  for presence of prefix before I run the entire function?
I didn't get you properly mentor!